tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -2168,6 +2168,66 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Translate litterals to there ScalaJS String representation. */
+  private def foldToStringForString_+(tree: Tree)(implicit pos : Position): Tree = tree match {
+    case FloatLiteral(value)   => StringLiteral(jsNumberToString(value.toDouble))
+    case DoubleLiteral(value)  => StringLiteral(jsNumberToString(value))
+    case LongLiteral(value)    => StringLiteral(value.toString)
+    case IntLiteral(value)     => StringLiteral(value.toString)
+    case BooleanLiteral(value) => StringLiteral(value.toString)
+    case Null()                => StringLiteral("null")
+    case Undefined()           => StringLiteral("undefined")
+
+    case _ => tree
+  }
+
+  /* Following the ECMAScript 6 specification */
+  private def jsNumberToString(value: Double): String = {
+    @inline def getJVMnks : (Int, Int, BigInt) = {
```

Comment:
No space before `:` (it's a thing of the French language to have spaces before colons)